snippets/product-media-gallery.liquid

```diff
@@ -72,7 +72,20 @@
           data-media-id="{{ section.id }}-{{ featured_media.id }}"
         >
           {%- assign media_position = 1 -%}
-          {% render 'product-thumbnail', media: featured_media, media_count: media_count, position: media_position, desktop_layout: section.settings.gallery_layout, mobile_layout: section.settings.mobile_thumbnails, loop: section.settings.enable_video_looping, modal_id: section.id, xr_button: true, media_width: media_width, lazy_load: false %}
+          {% render 'product-thumbnail',
+            media: featured_media,
+            media_count: media_count,
+            position: media_position,
+            desktop_layout: section.settings.gallery_layout,
```

Comment:
When both `constrain media to screen size` is `true` and `media fit` set to `fill`, the main first image when the layout is anything by two columns can get quite zoomed in. 
<details>
<summary>screenshot</summary>

![](https://screenshot.click/02-25-y0l66-vzdfk.png)
</details>

But I guess they more than enough options to work around it.